pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/DefaultIndexCreatorProvider.java

```diff
@@ -88,17 +90,25 @@ public ForwardIndexCreator newForwardIndexCreator(IndexCreationContext.Forward c
             context.getMaxRowLengthInBytes());
       }
     } else {
-      if (context.getFieldSpec().isSingleValueField()) {
-        if (context.isSorted()) {
-          return new SingleValueSortedForwardIndexCreator(context.getIndexDir(), context.getFieldSpec().getName(),
-              context.getCardinality());
+      if (context.forwardIndexDisabled()) {
```

Comment:
(nit) not related to this PR -- do you mind adding a comment under the else branch above line 93 ? something like `Dictionary enabled column`. Will make it bit more readable. Similarly above line 79.

We may also want to make this more readable by having separate helper method but separate PR for that is fine too

Reply:
done, will do the helper method changes as a follow up to avoid confusion in this PR. :)